### ISSUE TITLE
Doesn't reload modules

### ISSUE THREAD

**Issue by ukch** [samuelcolvin/watchgod#8]
I was hoping to use this library to build a development server that restarts itself whenever it detects a code change, in much the same way that Django or Flask do. Unfortunately, the following two problems prevent it from functioning as I would expect:

1. The first time I change a file, the server appears to restart as expected. Unfortunately, no code is reloaded, meaning my changes are not applied.
2. The second time I change a file, the change is not picked up on at all.

**Comment by samuelcolvin** [comment]
I think this is an error in your code, perhaps related to deep reloading. 

Take a look at [aiohttp-devtools](https://github.com/aio-libs/aiohttp-devtools) which I also built and uses watchgod for live reloading.

**Comment by samuelcolvin** [comment]
I assume this resolved your problem.

If you still have a problem, feel free to comment.

**Comment by ukch** [comment]
I have got something working using `subprocess`, which seems to solve my problem for now, thanks.

**Comment by samuelcolvin** [comment]
could be the same as aio-libs/aiohttp-devtools#181

**Comment by akvadrako** [comment]
What is one supposed to do to enable module reloading with watchdog?

**Comment by samuelcolvin** [comment]
probably the simplest solution, without looking at your code, is to use `run_process` or `subprocess` .

**Comment by akvadrako** [comment]
I am using `run_process` but it doesn't reload modules, so I'm wondering what the requirements are to get it to work.

As a workaround I am using `os.execl` like this:

```py
    def execmain():
        os.execl(sys.executable, sys.executable, '-m', 'src.main')

    run_process('.', execmain, args=())
```

**Comment by samuelcolvin** [comment]
well `run_process` reloads the python process, so it does reload modules.

I think the problem you have here is that you're not watching the files you want to watch, that's not the problem reported in this issue.

I guess files must be changing outside `.`, if that's the case you need to watch those directories too.

**Comment by akvadrako** [comment]
No, watchgod does see the changed files and does a reload, but the modules are still running the old code.

**Comment by samuelcolvin** [comment]
Then it's not a problem with watchgod.

The problem is likely because you are loading the module before watchgod calls your function in a different process, so python is forking the process and always using the originally loaded module.

This is the documented behaviour, and not related to watchgod.

**Comment by akvadrako** [comment]
That is the requirement I was asking for. This is related to watchgod because it's how it's designed – to take a function and use forking. So of course that function is loaded before watchgod runs.

It would be helpful to mention that in the `run_process` docs.

**Comment by samuelcolvin** [comment]
fixed by #92
